firebase-common/src/commonTest/kotlin/dev/gitlive/firebase/EncodersTest.kt

```diff
@@ -71,17 +71,19 @@ class EncodersTest {
     @Test
     fun testEncodeDecodedSealedClass() {
         val test = SealedClass.Test("Foo")
-        val encoded = encode(test, false)
-        val decoded = decode(encoded) as? SealedClass.Test
+        val serializer = SealedClass.Test.serializer() // has to be used because of JS issue
```

Comment:
legacy JS is not finding a serializer in `serializer<T>()` call (see [issue](https://github.com/Kotlin/kotlinx.serialization/issues/1448)) and it seems unlikely to be fixed. the upstream repo is using legacy JS and not IR JS, so I don't consider migrating at the moment